runtime/j9vm/jvm.c

```diff
@@ -2239,15 +2238,12 @@ JNI_CreateJavaVM_impl(JavaVM **pvm, void **penv, void *vm_args, BOOLEAN isJITSer
 #endif /* defined(J9UNIX) || defined(J9ZOS390) */
 
 	args = (JavaVMInitArgs *)vm_args;
-	launcherArgumentsSize = initialArgumentScan(args, &specialArgs);
-	localVerboseLevel = specialArgs.localVerboseLevel;
-
 #if defined(J9VM_OPT_SNAPSHOTS)
-	if (NULL != specialArgs.vmSnapshotFilePath) {
-		createParams.flags |= J9_CREATEJAVAVM_SNAPSHOT;
-		createParams.vmSnapshotFilePath = specialArgs.vmSnapshotFilePath;
-	}
+	launcherArgumentsSize = initialArgumentScan(args, &specialArgs, &createParams);
+#else /* defined(J9VM_OPT_SNAPSHOTS) */
+	launcherArgumentsSize = initialArgumentScan(args, &specialArgs);
 #endif /* defined(J9VM_OPT_SNAPSHOTS) */
+	localVerboseLevel = specialArgs.localVerboseLevel;
 
 	if (VERBOSE_INIT == localVerboseLevel) {
```

Comment:
I was going to mention it. But, you already figured it out.

Don't know if there is a historical reason for just updating `specialArgs` in `initialArgumentScan`. After reading the code, I don't see a reason that might stop us from passing `createParams` as an input argument to `initialArgumentScan`.

To avoid the `J9VM_OPT_SNAPSHOTS` ifdefs, we can move this `if` block inside `initialArgumentScan`.